### PR TITLE
basePath & servers support; support more schema type 

### PR DESCRIPTION
**Pull request by DDDHuang**
fixed: https://github.com/inaka/cowboy_swagger/issues/134

support more schema, as like open api 3.0.0:
```erlang
#{
        allOf => [
            #{
                type => object,
                required => [name],
                properties => #{
                    name => #{
                        type => string,
                        example => "exmaple"
                    }
                }
            },
            #{
                oneOf => [ cowboy_swagger:schema(<<"password_based">>)
                                 , cowboy_swagger:schema(<<"jwt">>)
                                 , cowboy_swagger:schema(<<"scram">>)
                ]    
            }
        ]
    },

```

### CODE REVIEW

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]
Hi @DDDHuang. Thanks for the PR.
A few requests:

1. Can we use the full word `translate` instead of just `trans` when possible?
2. Can you add some tests?

File: src/cowboy_swagger.erl [hunk 1]
```diff
@@ -246,12 +277,22 @@ normalize_list_values(List) ->
...
+  BasePath = deconstruct_openapi_url(GlobalSpec),
+  SwaggerPaths = swagger_paths(SanitizeTrails, BasePath),
+  GlobalSpec#{paths => SwaggerPaths};
 create_swagger_spec(GlobalSpec, SanitizeTrails) ->
   create_swagger_spec(GlobalSpec#{openapi => <<"3.0.0">>}, SanitizeTrails).
 
+%% @private
+deconstruct_openapi_url(GlobalSpec) ->
+    Servers = maps:get(servers, GlobalSpec, [#{}]),
+    Url = maps:get(url, hd(Servers), <<"">>),
```
Comment (elbrujohalcon) [inline src/cowboy_swagger.erl:293]:
```suggestion
    [Server|_] = maps:get(servers, GlobalSpec, [#{}]),
    Url = maps:get(url, Server, <<"">>),
```

Comment (elbrujohalcon) [inline src/cowboy_swagger.erl:293]:
```suggestion
    [Server|_] = maps:get(servers, GlobalSpec, [#{}]),
    Url = maps:get(url, Server, <<"">>),
```

File: test/cowboy_swagger_SUITE.erl [hunk 2]
```diff
@@ -173,6 +177,42 @@ schema_test(_Config) ->
...
+  ok = cowboy_swagger:add_definition(Definition2),
+
+  %%
+  %% Then
+  %%
+  {ok, SwaggerSpec1} = application:get_env(cowboy_swagger, global_spec),
+  JsonDefinitions = cowboy_swagger:get_existing_definitions(SwaggerSpec1),
+  true = maps:is_key(Name1, JsonDefinitions),
+  true = maps:is_key(Name2, JsonDefinitions),
+  ok.
```
Comment (elbrujohalcon) [inline test/cowboy_swagger_SUITE.erl:214]:
```suggestion
  {comment, ""}.
```
If your test case uses `ct:comment/1,2`, it must return `{comment, ""}` when it's successful.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]
Two small ammendments.

**Review by elbrujohalcon — APPROVED** [review]

**Comment by elbrujohalcon** [comment]
@DDDHuang Thanks! If you have any interest at all in becoming a maintainer of this repo, let me know and I can let you in. That way you'll be able to generate releases and publish them to hex.pm, among other things.

**Comment by DDDHuang** [comment]
I'm happy to be the maintainer.

**Comment by elbrujohalcon** [comment]
Well… as soon as you accept the invite, you'll be!
Ping me on slack if there is any issue.

**Pull request merged**